OgreMain/src/OgreResourceManager.cpp

```diff
@@ -344,24 +338,18 @@ namespace Ogre {
     {
         OGRE_LOCK_AUTO_MUTEX;
 
-        ResourceMap::iterator i, iend;
-        iend = mResources.end();
-        for (i = mResources.begin(); i != iend; )
+        for (auto& r : mResources)
```

Comment:
this will break, as remove() invalidates the iterator

Reply:
reverted